### PR TITLE
Support Windows

### PR DESCRIPTION
**Pull request by mattn**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -55,7 +55,7 @@ func v2() {
 		log.Fatalf("failed to check mpd supported functions: %v", err)
 	}
 	// get music dir from local mpd connection
-	if config.MPD.Network == "unix" && config.MPD.MusicDirectory == "" {
+	if config.MPD.IsLocal() && config.MPD.MusicDirectory == "" {
```
Comment (meiraka) [inline main.go:58]: config command probably only works with unix sockets.
https://github.com/MusicPlayerDaemon/MPD/blob/256cfc545ddf05ed72784922f3ec038da6747032/src/command/OtherCommands.cxx#L372-L376
https://github.com/MusicPlayerDaemon/MPD/blob/a2f5a63bbc1a21be672d358a107a3a206c0c5369/src/client/Client.hxx#L149-L152

Comment (mattn) [reply main.go:58]: Windows does not support UNIX domain socket. (Strictly speaking, Windows supports AF_UNIX, but mpd does not) So the `Network` is always `tcp`. On Windows, vv should check the address whether the network is tcp or not.

Comment (mattn) [reply main.go:58]: If vv check `unix` always, Windows does not have a way to get  music_directory.

Comment (meiraka) [reply main.go:58]: I think Windows mpd can't return music_directory.
If you change the condition of the if statement here to config.MPD.MusicDirectory == "", cl.Config() should return an error.

Comment (meiraka) [reply main.go:58]: I don't have Windows, so please tell me the mpd config command in a Windows environment.
I tried the config command using WINE, but it returned an error.

server (wine-4.0.3, mpd-0.20.23)
```
> wine mpd.exe --stdout --verbose mpd.conf
002f:err:msvcrt:MSVCRT__invalid_parameter (null):0 (null): (null) 0
config_file: loading file mpd.conf
vorbis: Xiph.Org libVorbis 1.3.6
opus: libopus 1.3
simple_db: reading DB
curl: version 7.61.1
exception: Input plugin 'ffmpeg' is unavailable: No protocol
client: [0] opened from 127.0.0.1:37018
client: [0] process command "config"
client: [0] command returned 2
```
client
```
> telnet localhost 6600
Trying 127.0.0.1...
Connected to localhost.
Escape character is '^]'.
OK MPD 0.20.22
config
ACK [4@0] {config} Command only permitted to local clients
```

Comment (mattn) [reply main.go:58]: I could make sure mpd return music_directory on Windows in few days ago. The motivation of changing this part is that music_directory was empty string on Windows. And vv is working fine on Windows with this change.

Comment (mattn) [reply main.go:58]: FYI, you can download v0.21.25 from here https://www.musicpd.org/download.html

Comment (meiraka) [reply main.go:58]: Did it work without changing the ParseConfig?
https://github.com/meiraka/vv/blob/fb6672f8c046e754d1a6a7288a2aa6bcd4b6e679/config.go#L60-L64

Comment (mattn) [reply main.go:58]: I can't remember clearly, so I'll reply about this in tonight.

**Pull request closed without merging**